docs/001_develop/03_client-capabilities/014_utility-methods/06_decorators-util.mdx

```diff
@@ -0,0 +1,91 @@
+---
+title: 'Decorators Utilities'
+sidebar_label: 'Decorators Utilities'
+id: decorators-utils
+keywords: [utils, utility, data, decorators, env, environment, formatters, number, date, logger, logging, mappers, mixins, observers, resource, server, serializer, json, styles, css, UUID, unique, window, iframe]
+tags:
+  - utils
+  - utility
+  - data
+  - decorators
+  - env
+  - environment
+  - formatters
+  - number
+  - date
+  - logger
+  - logging
+  - mappers
+  - mixins
+  - observers
+  - resource
+  - server
+  - serializer
+  - json
+  - styles
+  - css
+  - UUID
+  - unique
+  - window
+  - iframe
+---
+
+# Decorators utilities
+
+Provides utility decorators to enhance component behavior and development efficiency within TypeScript projects. Specifically, the `renderOnChange` decorator automates the re-rendering process of components upon property changes, leveraging the FAST framework for web development.
+
+## Key Features
+
+- **Automatic Rerendering:** Automatically invokes the `render()` method of a component when any decorated property changes.
+- **Integration with FAST Framework:** Designed to work seamlessly with components extending `FASTElement`.
+
+## Examples
+
+### Using `renderOnChange`
```

Comment:
Question: what is the difference between using this and an observable?

Reply:
from the PR that added this:
_renderOnChange decorator to remove some repetitive internal-observation boilerplate._